main/streams/streams.c

```diff
@@ -2332,6 +2332,27 @@ PHPAPI int _php_stream_scandir(const char *dirname, zend_string **namelist[], in
 }
 /* }}} */
 
+/* {{{ php_stream_is_cacheable
+ */
+PHPAPI int _php_stream_is_cacheable(const char *path, int options
+	, php_stream_context *context)
+{
+	php_stream_wrapper *wrapper = NULL;
+
+	if (!path || !*path) {
+		return 0;
+	}
+
+	wrapper = php_stream_locate_url_wrapper(path, NULL, options);
+
+	if (wrapper && wrapper->wops->stream_is_cacheable) {
+		return wrapper->wops->stream_is_cacheable(wrapper,
+				path, options, context);
+
+	} else return 0;
```

Comment:
Should the result go to an intermediate var, and then return this var ?

 

De : Reeze Xia [mailto:notifications@github.com] 
Envoyé : dimanche 4 janvier 2015 14:26
À : php/php-src
Cc : Francois Laupretre
Objet : Re: [php-src] Add is_cacheable() stream wrapper operation (#976)

 

In main/streams/streams.c <https://github.com/php/php-src/pull/976#discussion-diff-22439135> :

> +PHPAPI int _php_stream_is_cacheable(const char *path, int options
> +    , php_stream_context *context)
> +{
> +    php_stream_wrapper *wrapper = NULL;
> +
> +    if (!path || !*path) {
> +           return 0;
> +    }
> +
> +    wrapper = php_stream_locate_url_wrapper(path, NULL, options);
> +
> +    if (wrapper && wrapper->wops->stream_is_cacheable) {
> +           return wrapper->wops->stream_is_cacheable(wrapper,
> +                           path, options, context);
> +
> +    } else return 0;

This seems not consist with coding standards.

if (wrapper && wrapper->wops->stream_is_cacheable) {
    return wrapper->wops->stream_is_cacheable(wrapper,
                           path, options, context);
}
 
return 0;

—
Reply to this email directly or view it on GitHub <https://github.com/php/php-src/pull/976/files#r22439135> .Image supprimée par l'expéditeur.